auth/services/irma/validator.go

```diff
@@ -0,0 +1,311 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package irma
+
+import (
+	"crypto"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/lestrrat-go/jwx/jwt"
+	"github.com/nuts-foundation/go-did"
+	nutsCrypto "github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/vdr"
+	"github.com/nuts-foundation/nuts-node/vdr/types"
+	"time"
+
+	"github.com/nuts-foundation/nuts-node/auth/logging"
+
+	irmaserver2 "github.com/privacybydesign/irmago/server/irmaserver"
+
+	"github.com/nuts-foundation/nuts-node/auth/services"
+
+	"github.com/nuts-foundation/nuts-node/auth/contract"
+
+	irma "github.com/privacybydesign/irmago"
+	irmaserver "github.com/privacybydesign/irmago/server"
+)
+
+// todo rename to verifier
```

Comment:
todo? >> issue

Reply:
was already there, not sure if still relevant